drivers/storage/portworx/component/autopilot.go

```diff
@@ -265,26 +305,138 @@ func (c *autopilot) createConfigMap(
 }
 
+// createServiceAccountForOCP creates service account for OCP Prometheus
+// autopilot to access Prometheus metrics
+// This is required for OCP 4.16 and above
+func (c *autopilot) createServiceAccountForOCP(namespace string, ownerRef *metav1.OwnerReference) error {
+	return k8sutil.CreateOrUpdateServiceAccount(
+		c.k8sClient,
+		&v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            AutopilotPrometheusServiceAccountName,
+				Namespace:       namespace,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+		},
+		ownerRef,
+	)
+}
+
+// createClusterRoleBindingForOCP creates cluster role binding for OCP Prometheus
+// autopilot to access Prometheus metrics requires cluster-monitoring-view role
+// This is required for OCP 4.16 and above
+func (c *autopilot) createClusterRoleBindingForOCP(namespace string) error {
+	return k8sutil.CreateOrUpdateClusterRoleBinding(
+		c.k8sClient,
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: AutopilotPrometheusClusterRoleBindingName,
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     OpenshiftClusterViewRoleName,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      AutopilotPrometheusServiceAccountName,
+					Namespace: namespace,
+				},
+			},
+		},
+	)
+}
+
 func (c *autopilot) createSecret(clusterNamespace string, ownerRef *metav1.OwnerReference) error {
 
-	token, cert, err := c.getPrometheusTokenAndCert()
+	ocp416Plus, err := pxutil.IsSupportedOCPVersion(c.k8sClient, pxutil.Openshift_4_16_version)
 	if err != nil {
 		return err
 	}
 
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            AutopilotSecretName,
+			Namespace:       clusterNamespace,
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+		},
+	}
+
+	if ocp416Plus {
+		err = k8sutil.GetSecret(c.k8sClient, AutopilotSecretName, clusterNamespace, secret)
+		if err != nil {
+			return fmt.Errorf("Error during getting secret %v ", err)
+		}
+
+		if secret.Data != nil {
+			// if secret exists, check secret type
+			if secret.Type == v1.SecretTypeServiceAccountToken {
+				// if secret type is service account token, check if token is expired
+				ok, err := isTokenRefreshRequired(secret)
+				if err != nil {
+					return fmt.Errorf("Error during checking token refresh %v ", err)
+				}
+				if ok {
+					// refresh the token if it is expired
+					logrus.Infof("refreshing token %s in namespace %s", AutopilotSecretName, clusterNamespace)
+					err := c.refreshTokenSecret(secret, clusterNamespace, ownerRef)
+					if err != nil {
+						return fmt.Errorf("Error during refreshing token %v ", err)
+					}
+
+					// delete autopilot deployment to sync with new token
+					logrus.Info("deleting autopilot pod to sync with new token")
+					err = k8sutil.DeletePodsByLabel(c.k8sClient, map[string]string{"name": "autopilot"}, clusterNamespace)
```

Comment:
No, the secret is not mounted inside the pod, instead it is used as volume mount for token and cert, so it doesn't update the token after it is refreshed. Hence this is required.

Reply:
This can be handled using **ProjectedVolumeSource** as ColumeSource in autopilot deployment, it will update the token inside the pod without restart.
@zoxpx does autopilot needs to do anything everytime token is updated inisde the pod and pod is not restarted?